esrally/client.py

```diff
@@ -25,16 +25,15 @@ def __init__(self, hosts, client_options):
         # we're using an SSL context now and it is not allowed to have use_ssl present in client options anymore
         if self.client_options.pop("use_ssl", False):
             import ssl
-            from elasticsearch.connection import create_ssl_context
             self.logger.info("SSL support: on")
             self.client_options["scheme"] = "https"
 
-            self.ssl_context = create_ssl_context(cafile=self.client_options.pop("ca_certs", certifi.where()))
+            self.ssl_context = ssl.create_default_context(ssl.Purpose.CLIENT_AUTH, cafile=self.client_options.pop("ca_certs", certifi.where()))
 
             if not self.client_options.pop("verify_certs", True):
                 self.logger.info("SSL certificate verification: off")
-                self.ssl_context.check_hostname = False
                 self.ssl_context.verify_mode = ssl.CERT_NONE
+                self.ssl_context.check_hostname = False
```

Comment:
This is mandatory. When specifying `check_hostname` the verify_mode needs to be set already otherwise:

```
ValueError: check_hostname needs a SSL context with either CERT_OPTIONAL or CERT_REQUIRED
```